drivers/gpu/drm/lima/lima_gem.c

```diff
@@ -53,10 +55,15 @@ static struct lima_bo *lima_gem_create_bo(struct drm_device *dev, u32 size, u32
 	if (err)
 		goto err_out0;
 
-	bo->cpu_addr = dma_alloc_coherent(dev->dev, size, &bo->dma_addr, GFP_USER);
-	if (!bo->cpu_addr) {
-		err = -ENOMEM;
-		goto err_out1;
+	if (dma_addr)
```

Comment:
What about the bo free case, we can't do dma_free for bo imported from other device.

Reply:
True, do you have a suggestion on how to handle this? Is there some readily available reference counting that we can use to free this, or do we need to store some additional information in lima_bo to handle this?

Reply:
From the vc4_free_object of vc4 driver, we can use the gem_bo->import_attach to distinguish a self created bo and a imported bo.

Reply:
There's another place which is not shown in this patch that the bo->resv should also be from the dma_buf obj. Please reference the vc4_prime_import_sg_table. This resv is used to sync dma_buf read/write between multi users. In the display/render case, render will queue a write fence to this resv obj, display will wait the resv obj (so the write fence) before display. So the lima_bo should use the same resv as dma_buf.